src/wallets/injected/InjectedMathWallet.ts

```diff
@@ -0,0 +1,44 @@
+import { PublicKey, Signature } from "near-api-js/lib/utils/key_pair";
+
+interface MathAccount {
+  networkUnique: string;
+  address: string;
+  name: string;
+  authority: string;
+  selected: boolean;
+  type: string;
+
+  accountId: string;
+  publicKey: string;
+  permission: string;
+  network: string;
+}
+
+interface MathNetwork {
+  id: string;
+  name: string;
+  blockchain: string;
+  chainId: string;
+  nodeAddress: string;
+  nodeAddress2: string;
+  nodeAddress3: string;
+  enable: boolean;
+  custom: boolean;
+  extra: string;
+}
+
+export interface InjectedMathWallet {
+  signer: {
+    account: MathAccount;
+    network: MathNetwork;
+    createKey(accountId: string, networkId?: string): Promise<PublicKey>;
+    getPublicKey(accountId?: string, networkId?: string): Promise<PublicKey>;
+    signMessage(
+      message: Uint8Array,
+      accountId?: string,
+      networkId?: string
+    ): Promise<Signature>;
+  };
+  login: (param: object) => Promise<MathAccount>;
```

Comment:
Yeah looking at the docs, it seems there's two params: `publicKey` and `contractId`. Not entirely sure if `publicKey` is a string or the `PublicKey` type 😩

Reply:
In the first place I did not specify the params because I was not sure why they were used for and their types, in the next push the login method definition will be something like this:

```ts
login: (param: {
  publicKey?: PublicKey;
  contractId?: string;
}) =
```> Promise<MathAccount>;